dids/src/main/kotlin/web5/sdk/dids/methods/dht/DidDht.kt

```diff
@@ -202,6 +215,47 @@ public sealed class DidDhtApi(configuration: DidDhtConfiguration) {
     return BearerDid(didUri, did, keyManager, didDocument)
   }
 
+  /**
+   * Updates an existing "did:dht" DID by applying the provided changes to the DID Document.
+   *
+   * @param bearerDid The existing "did:dht" DID to update.
+   * @param options Optional parameters ([UpdateDidDhtOptions]) to specify additional keys, services, and optional
+   * publishing during the update.
+   * @return The updated [BearerDid] instance.
+   */
+  public fun update(bearerDid: BearerDid, options: UpdateDidDhtOptions): BearerDid {
+    val existingDidDocument = bearerDid.document
+
+    val updatedServices = options.services ?: existingDidDocument.service
+    val updatedControllers = options.controllers ?: existingDidDocument.controller
+    val updatedAlsoKnownAses = options.alsoKnownAses ?: existingDidDocument.alsoKnownAs
```

Comment:
I think you might have a bug here with how you're using the Elvis operator (👑 🎸) (`?:`). The `UpdateDidDhtOptions` has docs:

```kotlin
 * @property services A list of [Service]s to add to the DID Document.
// ...
 * @property controllers A list of controller DIDs to add to the DID Document.
 * @property alsoKnownAses A list of also known as identifiers to add to the DID Document.
```

As in, those values are additive, but the Elvis operator is a null coalescing operation, so imagine the `existingDidDocument.service` has service A, and then `options.services` has service B & C, then this would update the did:dht's DID Document to now have only services B & C.

Reply:
@KendallWeihe ah - that is a bug in the docs, it isn't adding now, but replacing (ie code is right, comments are not). will fix. thanks!

Reply:
Hmm, is the desired behavior to replace completely or add a new service to an existing list?

If it's the former, is this because you are prioritizing immutability?

I am not certain that going with former approach is a good dx. If I just want to add a service, shouldn't the update method do the work internally of:

1. check if service method is empty
2. if not empty, turn list mutable
3. add to list
4. make the list immutable again

?

Reply:
yeah desired is to entirely replace it (at least as it stands currently) which preserves immutability but keeps it relatively simple (but alternative means having add/remove etc things explicitly in there for things that are lists, perhaps add/remove for each thing you may want to change and separate methods?)

Reply:
> yeah desired is to entirely replace it

I agree with this approach, if only for the sake of pragmatism ("crawl before walk")